validation-tool-backend/src/review/review.service.ts

```diff
@@ -270,4 +271,73 @@ sum(case when gm.generic_metadata_id is not null then 1 else 0 end) as assigned_
       }
     }
   }
+
+  async getProgress(
+    run_ids: number[]
+  ): Promise<ReviewProgressDto[]> {
+    let response: ReviewProgressDto[] = undefined;
+    const pool = await this.databaseService.getConnection();
+    let sql_run_ids = run_ids.join(',');
+    
+    const result = await pool
+      .request()
+    //  .input('RunIDs', VarChar, run_ids.join(','))
+      .query(`WITH r AS 
+      (SELECT  
+        CASE WHEN is_dataset_reference IS NOT NULL AND agency_dataset_identified IS NOT NULL THEN 1 ELSE 0 END AS revd, 
+        d.run_id AS run_id
+      FROM dyad d LEFT JOIN snippet_validation sv ON (sv.dyad_id = d.id)
+      WHERE d.alias_id IS NOT NULL AND snippet IS NOT NULL
+      )
+     SELECT run_id, SUM(revd) as n_revd, COUNT(*) AS n_tot, SUM(revd)/(COUNT(*)*1.0)*100 as pct_complete
+      FROM r
+      WHERE run_id IN (${sql_run_ids})
+      GROUP BY run_id
+      ORDER BY run_id;`);
+    if (result?.recordset.length ) {
+      response = result.recordset;
+    }
+    return response;
+  }
+
+  async getStatistics(
+    run_ids: number[]
+  ) {
```

Comment:
any reason this is not typed like the other call?

Reply:
Fixed